complearn-genbank/routes/redis.ts

```diff
@@ -22,55 +22,126 @@ connectToRedis();
-        res.status(500).json({ error: 'Redis operation failed' });
+interface RedisGetRequest {
+    key: string;
+}
+
+interface RedisSetRequest {
+    key: string;
+    value: string | object;
+}
+
+interface RedisIncrRequest {
+    key: string;
+}
+
+interface RedisDelRequest {
+    key: string;
+}
+
+// Use Router type for route handlers
+router.post(
+    "/set",
+    (req: Request<{}, any, RedisSetRequest>, res: Response) => {
+        return new Promise<void>(async (resolve) => {
+            try {
+                const { key, value } = req.body;
+                if (!key || value === undefined) {
+                    res.status(400).json({ error: 'Key and value are required' });
+                    return resolve();
+                }
+
+                const stringValue = typeof value === 'object' ? JSON.stringify(value) : value;
+                await client.set(key, stringValue);
+                res.json({ success: true });
+                resolve();
+            } catch (error) {
+                console.error('Redis set error:', error);
+                res.status(500).json({
+                    error: 'Redis operation failed',
+                    details: error instanceof Error ? error.message : 'Unknown error'
+                });
+                resolve();
+            }
+        });
     }
-});
+);
 
+router.post(
+    "/get",
+    (req: Request<{}, any, RedisGetRequest>, res: Response) => {
+        return new Promise<void>(async (resolve) => {
+            try {
+                const { key } = req.body;
+                if (!key) {
+                    res.status(400).json({ error: 'Key is required' });
+                    return resolve();
+                }
 
-router.post("/set", async (req, res) => {
-    try {
-        const { key, value, ttl } = req.body;
-        if (ttl) {
-            await client.setEx(key, ttl, value);
-        } else {
-            await client.set(key, value);
-        }
-        res.json({ success: true });
-    } catch (error) {
-        logger.error('Redis set error:', error);
-        res.status(500).json({ error: 'Redis operation failed' });
+                const value = await client.get(key);
+                try {
+                    const parsedValue = value ? JSON.parse(value) : null;
+                    res.json({ value: parsedValue });
+                } catch {
+                    res.json({ value });
+                }
+                resolve();
+            } catch (error) {
+                console.error('Redis get error:', error);
+                res.status(500).json({
+                    error: 'Redis operation failed',
+                    details: error instanceof Error ? error.message : 'Unknown error'
+                });
+                resolve();
+            }
+        });
     }
-});
+);
 
+router.post(
+    "/incr",
+    (req: Request<{}, any, RedisIncrRequest>, res: Response) => {
+        return new Promise<void>(async (resolve) => {
```

Comment:
should not use new Promise
should declare this anonymous function `async` instead